webapp/app/js/compare.js

```diff
@@ -0,0 +1,494 @@
+ * 
+ * @param values
+ *        An array of numbers.
+ * @param avg
+ *        Average of the values.
+ * @return a number (the standard deviation)
+ */
+function stddev(values, avg) {
+  if (values.length <= 1) {
+    return 0;
+  }
+
+  return Math.sqrt(
+    values.map(function (v) { return Math.pow(v - avg, 2); })
+          .reduce(function (a, b) { return a + b; }) / (values.length - 1));
+}
+
+
+// -------------------------------------------------------------------------
+// End Utility Functions
+// -------------------------------------------------------------------------
+
+
+var comparetalos = angular.module("perf", ['ui.router', 'ui.bootstrap']);
+
+/* Copied from providers.js */
+comparetalos.provider('thServiceDomain', function() {
+    this.$get = function() {
+        if (window.thServiceDomain) {
+            return window.thServiceDomain;
+        } else {
+            return "";
+        }
+    };
+});
+
+comparetalos.factory('getSeriesSummary', [ function() {
+  return function(signature, signatureProps, optionCollectionMap, pgo, e10s) {
+    var platform = signatureProps.machine_platform + " " +
+      signatureProps.machine_architecture;
+    var extra = "";
+    if (signatureProps.job_group_symbol === "T-e10s") {
+      extra = " e10s";
+    }
+    var testName = signatureProps.test;
+    var subtestSignatures;
+    if (testName === undefined) {
+      testName = "summary";
+      subtestSignatures = signatureProps.subtest_signatures;
+    }
+    var name = signatureProps.suite + " " + testName +
+      " " + optionCollectionMap[signatureProps.option_collection_hash] + extra;
+
+    //Only keep summary signatures, filter in/out e10s and pgo
+    if (name.indexOf('summary') <= 0) {
+        return {};
+    }
+    if (e10s && (name.indexOf('e10s') <= 0)) {
+        return {};
+    } else if (!e10s && (name.indexOf('e10s') > 0)) {
+        return {};
+    }
+
+    //TODO: pgo is linux/windows only- what about osx and android
+    if (pgo && (name.indexOf('pgo') <= 1)) {
+        return {};
+    } else if (!pgo && (name.indexOf('pgo') > 0)) {
+        return {};
+    }
+
+    return { name: name, signature: signature, platform: platform,
+             subtestSignatures: subtestSignatures };
+  };
+}]);
+
+comparetalos.controller('CompareCtrl', [ '$state', '$stateParams', '$scope', '$rootScope', '$location',
+                              '$modal', 'thServiceDomain', '$http', '$q', '$timeout', 'getSeriesSummary',
+  function CompareCtrl($state, $stateParams, $scope, $rootScope, $location, $modal,
+                    thServiceDomain, $http, $q, $timeout, getSeriesSummary) {
+
+    var availableColors = [ 'red', 'green', 'blue', 'orange', 'purple' ];
+
+    function displayComparision() {
+//TODO: why do we need so much history?
+      $scope.timeRange = 2592000; // last 30 days
+      $scope.testList = [];
+
+//TODO: how to input these variables
+$scope.e10s = false;
+$scope.pgo = false;
```

Comment:
You can get them via the $stateParams variable, see line 473 in this file. https://github.com/mozilla/treeherder-ui/pull/465/files#diff-921250c6e3d3318db57e5548862c72faR473

P.S. You only want to define variables as members of $scope if you want to bind them back to the view, which AFAIK in this case you do not.